src/app/common/context/MustGatherContext.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import { useMustGathersQuery } from '@app/queries';
+import { UseQueryResult } from 'react-query';
+import { IMustGatherResponse, mustGatherStatus } from '@app/client/types';
+import { MustGatherWatcher } from '@app/common/components/MustGatherWatcher';
+import { AppContext } from '@app/common/context/AppContext';
+
+export type mustGatherObjType = {
+  customName: string;
+  type: 'plan' | 'vm';
+  status: mustGatherStatus;
+};
+
+export type mgWatchListType = {
+  name: string;
+  isGathering: boolean;
+}[];
+
+export type mustGatherListType = mustGatherObjType[];
+interface IMustGatherContext {
+  mustGatherModalOpen: boolean;
+  setMustGatherModalOpen: (isOpen: boolean) => void;
+  mustGatherList: mustGatherListType;
+  setActiveMustGather: (mustGatherObj: mustGatherObjType) => void;
+  activeMustGather: mustGatherObjType | undefined;
+  mustGatherWatchList: mgWatchListType;
+  setMustGatherWatchList: (list: mgWatchListType) => void;
+  mustGathers: UseQueryResult<IMustGatherResponse[], Response>;
```

Comment:
Ah, I didn't realize it comes all the way from up here. Per my earlier comment, consumer usage might be more readable if we called this `mustGathersQuery` or `mustGathersResult`.